runtime/src/chpl-gpu.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2020-2021 Hewlett Packard Enterprise Development LP
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.  * 
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * 
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This #define needs to be before the other #includes
+// since it affects included files
+#ifdef __linux__
+#define _GNU_SOURCE
+#endif
```

Comment:
I'm not following why we would need to set `_GNU_SOURCE` only for linux? What are you using that is a gnu extension? See also `runtime/include/sys_basic.h` which does this kind of thing and you could possibly include first here.

Reply:
There's no good reason I did that. I think I just ended up copying and pasting it from `chplsys.c` (where I first implemented these functions before factoring them into their own files). I guess I was thinking maybe it was necessary for including `chplrt.h` but it looks like that's not the case.

Is sys_basic.h generally an acceptable file to include in every .c file for the runtime.  It looks like if I include that and `chplrt.h` I get everything I need to compile `chpl-gpu.c`

Reply:
Yes I think you can include `sys_basic.h` if it meets your needs.